modules/storage-page-memory/src/test/java/org/apache/ignite/internal/storage/pagememory/index/hash/AbstractPageMemoryHashIndexStorageTest.java

```diff
@@ -89,4 +96,49 @@ void testFragmentedIndexColumns() {
 
         assertThat(getAll(index, createIndexRow(serializer, new RowId(TEST_PARTITION), 1, "foo")), empty());
     }
+
+    private static Stream<Arguments> sizesForCollisionTest() {
+        return Stream.of(
+                Arguments.of(10, 200),
+                Arguments.of(100, 200),
+                Arguments.of(200, 10),
+                Arguments.of(200, 100)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("sizesForCollisionTest")
+    void testHashCollisionAndDifferentSizes(int firstSize, int secondSize) {
+        var index = (PageMemoryHashIndexStorage) createIndexStorage(INDEX_NAME, ColumnType.STRING);
+
+        int hash = 123;
+
+        partitionStorage.runConsistently(locker -> {
+            try {
+                int inlineSize = index.indexTree().inlineSize();
+
+                HashIndexRow firstHashIndexRow = new HashIndexRow(
```

Comment:
It's just funny that you use `var` for closures but not for row.

Reply:
Closure classes have very long names, I didn't want to have multi-lines statement. Rows statements are already multi-lined. I don't use `var` by default for some reason.